src/DocumentDB.ChangeFeedProcessor/FeedProcessing/PartitionProcessor.cs

```diff
@@ -58,7 +58,15 @@ public async Task RunAsync(CancellationToken cancellationToken)
                         lastContinuation = response.ResponseContinuation;
                         if (response.Count > 0)
                         {
-                            await this.DispatchChanges(response, cancellationToken).ConfigureAwait(false);
+                            try
+                            {
+                                await this.DispatchChanges(response, cancellationToken).ConfigureAwait(false);
```

Comment:
Problem here is that DCE may come from checkpointing, and not from the actual users' code. So this needs to be done lower in stack.

Reply:
Agree, I refactored to make sure we only treat when exception comes from user code